app/assets/javascripts/discourse/widgets/post-cooked.js

```diff
@@ -41,14 +41,31 @@ export default class PostCooked {
   }
 
   init() {
-    const $html = this._computeCooked();
-    this._insertQuoteControls($html);
-    this._showLinkCounts($html);
-    this._fixImageSizes($html);
-    this._applySearchHighlight($html);
-
-    _decorators.forEach(cb => cb($html, this.decoratorHelper));
-    return $html[0];
+    const cookedDiv = this._computeCooked();
+    const $cookedDiv = $(cookedDiv);
+
+    this._insertQuoteControls($cookedDiv);
+    this._showLinkCounts($cookedDiv);
+    this._fixImageSizes($cookedDiv);
+    this._applySearchHighlight($cookedDiv);
+
+    this._decorateAndAdopt(cookedDiv);
+
+    return cookedDiv;
+  }
+
+  _decorateAndAdopt(cooked) {
+    const $cooked = $(cooked);
+
+    _decorators
+      .filter(d => !d.afterAdopt)
+      .forEach(d => d.callback($cooked, this.decoratorHelper));
```

Comment:
we want to move away from sending jquery object in decorate cooked, I understand you do this for consistency... but I'm unsure, maybe we should just give regular object? Wha do you think @eviltrout ?